src/utils/sessions.js

```diff
@@ -0,0 +1,60 @@
+import { getEmail, getPassphrase } from '../utils/encrypter';
+import miniLock from '../utils/miniLock';
+
+// TODO: will be different host from a mobile device, probably if (!window)
+const authUrl = `${window.location.origin}/api/login`;
+
+
+async function connectWithAuthRequest(initiateConnection, mID, secretKey, isNewPassphrase) {
+  const response = await fetch(authUrl, {
+    method: "GET",
+    headers: {
+      'X-Minilock-Id': mID
+    },
+  }
+  );
+  const message = await response.blob();
+  miniLock.crypto.decryptFile(message, mID, secretKey,
+    function (fileBlob, saveName, senderID) {
+      const reader = new FileReader();
+      reader.addEventListener("loadend", () => {
+        const authToken = reader.result;
+        initiateConnection({
+          authToken,
+          secretKey,
+          mID,
+          isNewRoom: isNewPassphrase,
```

Comment:
Step 4. Do the auth dance, get the token, and send that to the init connection action in redux.